File: Java Structure/fontRendering/TextMaster.java
class TextMaster:

    private static Map<FontType, List<GUIText>> texts = new HashMap<FontType, List<GUIText>>()

    def __init__(self, Loader theLoader):
        self.renderer = FontRenderer()
        self.loader = theLoader

    def render(self):
        renderer.render(texts)

    def loadText(self, text):
        font = text.getFont()   # (FontType) font
        data = font.loadText(text)  # (TextMeshData) data
        int vao = loader.loadToVAO(data.getVertexPositions(), data.getTextureCoords())
        text.setMeshInfo(vao, data.getVertexCount())
        textBatch = self.texts.get(font)
        if(textBatch == None):
            textBatch = []
            self.texts.append(font:textBatch)
        textBatch.append(text)

    def removeText(self, GUIText text):
        textBatch = self.texts.get(text.getFont())
        textBatch.remove(text)
        if(textBatch.isEmpty()):
            texts.remove(text.getFont())

    def cleanUp(self):
        self.renderer.cleanUp()



So... calling this font:
    Loader loader = new Loader();
    TextMaster.init(loader);
    FontType font = new FontType(loader.loadTexture("verdana"), new File("verdana.fnt"));
    # Parameters (
        text to render,
        font size,
        font,
        the position,
        line length -> 1.0 = width of screen,
        whether text is centered)
    GUIText text = new GUIText ("This is a test text!", 1, font, new Vector2f(0,0), 1.0, true);
    text.setColour(1,0,0);

File: Claver/assistant/avatar/fontRendering/TextMaster.py
from Claver.assistant.avatar.fontRendering.FontRenderer import FontRenderer
from Claver.assistant.avatar.fontMeshCreator.FontType import FontType
from Claver.assistant.avatar.fontMeshCreator.TextMeshData import TextMeshData

class TextMaster:

    __loader = None
    __texts = {}
    __renderer = None

    @staticmethod
    def init(theLoader):
        TextMaster.__renderer = FontRenderer()
        TextMaster.__loader = theLoader

    @staticmethod
    def render():
        TextMaster.__renderer.render(TextMaster.__texts)

    @staticmethod
    def loadText(text):
        font = text.getFont()
        data = font.loadText(text)
        vao = TextMaster.__loader.loadQuadToVAO(data.getVertexPositions(), data.getTextureCoords())
        text.setMeshInfo(vao, data.getVertexCount())
        if font in TextMaster.__texts:
            textBatch = TextMaster.__texts[font]
        else:
            textBatch = []
            TextMaster.__texts[font] = textBatch
        textBatch.append(text)      # Pass-by-reference means text is added to the dictionary

    @staticmethod
    def removeText(text):
        if text.getFont() in TextMaster.__texts:
            textBatch = TextMaster.__texts[text.getFont()]
            textBatch.remove(text)
            if not textBatch:
                del TextMaster.__texts[text.getFont()]
                # Also delete text's VAO and related VBOs?

    @staticmethod
    def cleanUp():
        TextMaster.__renderer.cleanUp()
